### DISCUSSION TITLE
Unable to transform response from server | base64-encoded response not parsed

### DISCUSSION THREAD

**Discussion by revmischa** [trpc/trpc#4033] [Q&A]
### Provide environment information

```
  System:
    OS: macOS 13.2
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 298.91 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 9.2.0 - /usr/local/bin/npm
  Browsers:
    Brave Browser: 111.1.49.120
    Chrome: 109.0.5414.87
    Firefox: 105.0.3
    Safari: 16.3
  npmPackages:
    @tanstack/react-query: ^4.24.10 => 4.24.10
    @trpc/client: ^10.14.0 => 10.14.0
    @trpc/next: ^10.14.0 => 10.14.0
    @trpc/react-query: ^10.14.0 => 10.14.0
    @trpc/server: ^10.14.0 => 10.14.0
    next: 13.2.4 => 13.2.4
    react: 18.2.0 => 18.2.0
    typescript: ^4.9.5 => 4.9.5
```

### Describe the bug

Everything works nicely locally. When I deploy my NextJS app to AWS using cdk-nextjs-standalone and query trpc I get the error: `TRPCClientError: Unable to transform response from server`

Looking at the response, it is a well-formatted JSON response in base64

The response is:
<img width="691" alt="Screenshot 2023-03-18 at 6 11 52 PM" src="https://user-images.githubusercontent.com/245131/226148425-fd3a585f-545f-4452-bced-16b877bbb8f2.png">

The body decodes to:
```json
[{"result":{"data":[{"id":"01GVVNZWCJ616C6NSD7AF8X0EC","accountName":"aaa","userId":"01GVVNZWCJPSKCCAZT963P54ZB","name":"aaa","tags":[],"items":[],"created":"2023-03-19T00:58:25.554Z","updated":"2023-03-19T00:58:25.554Z"}]}}]
```

The error is here. It expects `t.result` to be set but it's undefined. It looks like it didn't try to code the base64 response perhaps:
<img width="1045" alt="Screenshot 2023-03-18 at 6 12 49 PM" src="https://user-images.githubusercontent.com/245131/226148454-5f9487cd-2f5d-46fc-8196-36532bb76fa2.png">

### Link to reproduction

just using the most basic starter setup

### To reproduce

Deploy any simple tRPC app using https://github.com/jetbridge/cdk-nextjs/

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by KATT** [comment]
We need a step by step reproduction, ideally that doesn't require a deployment or an AWS account.

**Comment by revmischa** [comment]
> We need a step by step reproduction, ideally that doesn't require a deployment or an AWS account.

That's a reasonable request but unfortunately I don't know how to provide one. I was hoping someone might recognize this error or why it could occur so I could have some starting point for debugging the AWS deployment.

**Comment by KATT** [comment]
I am pretty sure this has to do with your deployment if you can't reproduce it in development

**Comment by KATT** [comment]
I have no idea why the `body` above is base64-encoded for instance.

**Comment by iduuck** [comment]
I assume, that this has something to do with the API Gateway (?!) – I just did a quick search on Google for `"isBase64Encoded"`. Seems like the API Gateway is encoding it somehow, and this is not a valid request being sent to tRPC.

Read more: https://aws.amazon.com/blogs/compute/handling-binary-data-using-amazon-api-gateway-http-apis/

**Comment by revmischa** [comment]
I don't have an API gateway, just NextJS.

**Comment by iduuck** [comment]
Just to be clear: There is a "problem" with your deployment, since Lambda seems to automatically use the previously linked binary data conversion.

@revmischa Do you have an example deployment, that we could possibly inspect? Because it is hard for OSS maintainer to "just deploy" to AWS (because this would result in costs, that are not feasable for OSS maintaining).

@KATT I don't know how to proceed here, since this is an issue with AWS, but *could* be handled by us in the client. However this is a very specific example and would potentially bother other users, too.

**Reply by KATT** [reply]
I converted it to a discussion as it seems very unlikely this is tRPC-related. If it proves to be wrong, we can convert it back

**Comment by revmischa** [comment]
Not caused by tRPC. Solved here: https://github.com/serverless-stack/open-next/commit/b4e096be329d237b2d40ef917d60d73ff997ed91

**Comment by spaceemotion** [comment]
I am getting the same error, also with my app hosted on AWS, and the issue does not seem to come from tRPC itself - but tRPC does not tell me what the server responded with.

According to Sentry, the response was a 500 error, but my server logs show no errors - so I am guessing it has to be something in between. For some reason tRPC does not include what the response actually is, so I am left with just the error code in Sentry. As far as I can tell, there are also no hooks or overwrites I can use to grab the response before the batch loader and then superjson are transforming the response?

**Comment by wodepig** [comment]
我是nuxt项目,包含nuxt ui, 解决方式是把根目录中的.npmrc文件删除就正常了. gpt告诉我说是shard路径扁平化的问题
I am a nuxt project, including nuxt ui, the solution is to delete the. npmrc file in the root directory.Gpt told me that it was a problem of shard path flattening.
